### PR TITLE
Add new module eos_l2_interfaces

### PR DESCRIPTION
**Pull request by Qalthos**
##### SUMMARY

Replaces ansible/network#13

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

eos_l2_interfaces

### CODE REVIEW

**Review by danielmellado — COMMENTED** [review]
There seems to be an issue with the tests, please check zuul output:

KeyError: 'transport'

Thanks!

**Comment by ikhan2010** [comment]
@Qalthos can you please look into it?  Let's get the CI green on this.

File: lib/ansible/module_utils/network/eos/config/base.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+The base class for all eos resource modules
+"""
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.module_utils.connection import Connection
+
+
+class ConfigBase(object):
```
Comment (ganeshrn) [inline lib/ansible/module_utils/network/eos/config/base.py]: This file is not required

**Comment by Qalthos** [comment]
recheck

File: lib/ansible/modules/network/eos/eos_l2_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+  state:
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    default: merged
+    description:
+    - The state the configuration should be left in
+    type: str
```
Comment (ganeshrn) [inline lib/ansible/modules/network/eos/eos_l2_interfaces.py:91]: Please add a notes section mentioning the eos version tested, connection type used and link to eos platform guide.
Refer https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L103

**Review by ganeshrn — APPROVED** [review]
~Other than the notes comments, lgtm~

File: lib/ansible/module_utils/network/eos/facts/l2_interfaces/l2_interfaces.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                    continue
+
+                has_allowed = re.match(r"allowed vlan (\S+)", match)
+                if has_allowed:
+                    # TODO: listify?
+                    trunk["trunk_allowed_vlans"] = has_allowed.group(1)
+                    continue
+            config['trunk'] = trunk
+
+        return utils.remove_empties(config)
```
Comment (ganeshrn) [inline lib/ansible/module_utils/network/eos/facts/l2_interfaces/l2_interfaces.py:101]: The logic does not validate if the facts are rendered in argspec format. Refer https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/ios/facts/interfaces/interfaces.py#L65

**Pull request merged**